### PR TITLE
Adds optional bandwidth profiling for wds connections

### PR DESCRIPTION
**Pull request by ibeckermayer**
This was helpful for making inferences during performance profiling.

### CODE REVIEW

File: lib/web/apiserver.go [hunk 1]
```diff
@@ -844,11 +844,11 @@ func (h *Handler) bindDefaultEndpoints() {
 	h.GET("/webapi/sites/:site/desktops", h.WithClusterAuth(h.clusterDesktopsGet))
 	h.GET("/webapi/sites/:site/desktopservices", h.WithClusterAuth(h.clusterDesktopServicesGet))
 	h.GET("/webapi/sites/:site/desktops/:desktopName", h.WithClusterAuth(h.getDesktopHandle))
-	// GET /webapi/sites/:site/desktops/:desktopName/connect?access_token=<bearer_token>&username=<username>&width=<width>&height=<height>
+	// GET /webapi/sites/:site/desktops/:desktopName/connect?access_token=<bearer_token>&username=<username>
```
Comment (flyinghermit) [inline lib/web/apiserver.go:847]: what purpose does this commented code serve?

File: lib/web/desktop.go [hunk 2]
```diff
@@ -426,10 +427,34 @@ func (c *connector) tryConnect(clusterName, desktopServiceID string) (conn net.C
 // connection to the browser (ws) and the mTLS connection to Windows
 // Desktop Serivce (wds)
 func proxyWebsocketConn(ws *websocket.Conn, wds net.Conn) error {
+	profileWdsBandwidth := false // Note to reviewers: this variable should always be false in production.
+	var start time.Time
+	var bytesReadFromWds int64
```
Comment (flyinghermit) [inline lib/web/desktop.go:432]: nit:
```suggestion
	var bytesReadFromWds, bytesReadFromWds int64
```

File: lib/web/desktop.go [hunk 3]
```diff
@@ -426,10 +427,34 @@ func (c *connector) tryConnect(clusterName, desktopServiceID string) (conn net.C
...
 	close := func() {
 		ws.Close()
 		wds.Close()
+		if profileWdsBandwidth {
+			duration := time.Since(start)
+			bytesReadPerSec := float64(bytesReadFromWds) / duration.Seconds()
+			bytesWrittenPerSec := float64(bytesWrittenToWds) / duration.Seconds()
+			totalBandwidth := float64(bytesReadFromWds+bytesWrittenToWds) / duration.Seconds()
+
+			log.Printf("TDP websocket closed. Read %v bytes from wds, wrote %v bytes to wds, in %v. Bytes read/sec: %v, bytes written/sec: %v, Total bandwidth over wds: %v",
```
Comment (flyinghermit) [inline lib/web/desktop.go:449]: nit: should we use structured logging here? `log.WithField().Debugf()`

File: lib/web/desktop.go [hunk 4]
```diff
@@ -426,10 +427,34 @@ func (c *connector) tryConnect(clusterName, desktopServiceID string) (conn net.C
 // connection to the browser (ws) and the mTLS connection to Windows
 // Desktop Serivce (wds)
 func proxyWebsocketConn(ws *websocket.Conn, wds net.Conn) error {
+	profileWdsBandwidth := false // Note to reviewers: this variable should always be false in production.
```
Comment (flyinghermit) [inline lib/web/desktop.go:430]: don't we have some global "debug/profiling" flag?

File: lib/web/desktop.go [hunk 5]
```diff
@@ -58,7 +59,7 @@ import (
 	"github.com/gravitational/teleport/lib/web/scripts"
 )
 
-// GET /webapi/sites/:site/desktops/:desktopName/connect?access_token=<bearer_token>&username=<username>
+// GET /webapi/sites/:site/desktops/:desktopName/connect/ws?username=<username>
```
Comment (flyinghermit) [inline lib/web/desktop.go:62]: I realize this is not a new addition introduced in this PR, but sending `username` in URL caught my eye. This just risks exposing username (especially email addresses) in unwanted places. Why not `userid`?

**Comment by zmb3** [comment]
Personally, I don't think we need to merge this. If we want to do similar tests in the future, it's only a 35 line change and is probably pretty easy to recreate. We can keep the branch around if you want, or just save a patch file somewhere.

**Pull request closed without merging**